### PR TITLE
Remove Node 12, use Node 16 as default engine, add Node 18 to build

### PR DESCRIPTION
**Pull request by jimmycallin**
### Reasons for making this change

Node 16 is current LTS version, and Node 12 has been deprecated for some time now.

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [ ] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/utils/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
   ],
   "engineStrict": false,
   "engines": {
-    "node": ">=12"
+    "node": ">=14"
```
Comment (heath-freenome) [inline packages/utils/package.json:13]: Whoops, looks like the `validator-ajv6` is missing the `engines` block. Would you be willing to add it?

**Review by heath-freenome — COMMENTED** [review]
@jimmycallin One little addition and this looks good!

File: packages/utils/package.json (hunk 1 above)
Comment (jimmycallin) [reply packages/utils/package.json:13]: Done!

File: packages/validator-ajv6/package.json [hunk 2]
```diff
@@ -69,4 +73,4 @@
     "url": "git+https://github.com/rjsf-team/react-jsonschema-form.git"
   },
   "license": "Apache-2.0"
-}
+}
```
Comment (heath-freenome) [inline packages/validator-ajv6/package.json:76]: Restore the blank line at the end of this file an merge away!

**Review by heath-freenome — APPROVED** [review]

**Comment by heath-freenome** [comment]
@jimmycallin I'm curious how much time you have free to do some fun things like, maybe, finish converting the `core/widgets` directory to Typescript?

**Comment by jimmycallin** [comment]
my typescript is quite poor, but perhaps this is a good place to learn. i'll see what i can do, but no promises :)

**Comment by heath-freenome** [comment]
> my typescript is quite poor, but perhaps this is a good place to learn. i'll see what i can do, but no promises :)

I'm willing to spend an hour with you showing you the basics... We can use the [discord server](https://github.com/rjsf-team/react-jsonschema-form/issues/2417) to communicate too!

**Comment by jimmycallin** [comment]
how kind of you! i'll try to get started myself a bit first this weekend, and will you on discord if/when i need some guidance!

**Pull request merged**